src/formatDate.js

```diff
@@ -8,7 +8,26 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
+  const dateParts = date.split(fromSeparator);
+  const dateMap = fromFormat.reduce((map, part, index) => {
+    map[part] = dateParts[index];
+
+    return map;
+  }, {});
+
+  if (dateMap['YYYY'] && toFormat.includes('YY')) {
+    dateMap['YY'] = dateMap['YYYY'].slice(-2);
+  } else if (dateMap['YY'] && toFormat.includes('YYYY')) {
+    const year = parseInt(dateMap['YY'], 10);
+
+    dateMap['YYYY'] = year < 30 ? `20${dateMap['YY']}` : `19${dateMap['YY']}`;
+  }
+
+  const newDateParts = toFormat.map((part) => dateMap[part]);
```

Comment:
The mapping from `toFormat` to `newDateParts` assumes that all parts exist in `dateMap`. It would be safer to check if `dateMap[part]` is defined before accessing it to avoid potential undefined values in the output.